### PR TITLE
test: improve unit testing coverage

### PR DESCRIPTION
**Pull request by pstachel**
part of #1445

### CODE REVIEW

File: tests/unit/button-plain.spec.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
+import { shallowMount } from '@vue/test-utils';
+import ButtonPlain from '../../src/popup/router/components/ButtonPlain.vue';
+
+describe('ButtonPlain', () => {
+  it('should emit click event on click', async () => {
```
Comment (CedrikNikita) [inline tests/unit/button-plain.spec.js:5]: IMO, I don't think tha we should test that button is emiting click event, every other test is using button, and if the click would fail the test would fail as well.

**Review by CedrikNikita — COMMENTED** [review]
I also believe that there is mainly `component test` type presented here, but the issue was about unit tests.

https://vuejs.org/guide/scaling-up/testing.html

File: tests/unit/helper-functions/convert-token.spec.js [hunk 2]
```diff
@@ -0,0 +1,14 @@
+import { convertToken } from '../../../src/popup/utils/helper';
+
+const invalidInput = [null, 'test', undefined, {}, [], NaN, '1', '-1'];
+
+describe('convertToken', () => {
+  [2, -2, 0].forEach((precision) => it('should move decimal point n places to desired direction', () => {
+    const value = 1234.9876;
+    const expectedResult = value * (10 ** precision);
```
Comment (CedrikNikita) [inline tests/unit/helper-functions/convert-token.spec.js:8]: You've missed the point during creation of this test. We need to test that it converts with the maximum precision each digit should still be there. 
We need to test here edge cases something like 20 30 digits powering. Need to use BigInt or BigNumber here to test properly.

File: tests/unit/tab-bar.spec.js [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+describe('TabBar', () => {
+  routes.forEach((route, index) => it('should have assigned route', () => {
+    const wrapper = shallowMount(TabBar, {
+      stubs: {
+        RouterLink: RouterLinkStub,
+      },
+    });
+    expect(wrapper.findAllComponents(RouterLinkStub).at(index).props().to).toStrictEqual(route);
+  }));
+  it('should have same amount of routes and RouterLinks', () => {
```
Comment (CedrikNikita) [inline tests/unit/tab-bar.spec.js:20]: In this case we would need to change this test everytime we would decide to change the routes in the component.

File: src/popup/router/components/AccountCard.vue [hunk 4]
```diff
@@ -28,8 +28,8 @@ export default {
   },
   props: {
     idx: { type: Number, required: true },
-    color: { type: String, required: true },
-    shift: { type: Number, required: true },
+    color: { type: String },
```
Comment (CedrikNikita) [inline src/popup/router/components/AccountCard.vue:31]: The `color` and `shift` is actually required, why do you decide to change that.

Comment (pstachel) [reply src/popup/router/components/AccountCard.vue:31]: Because when `AccountCard` is used somewhere it always receive all of the props through `v-bind` and it resulted in errors in the tests

**Review by CedrikNikita — APPROVED** [review]
